modules/RTC/ortc/RTCPeerConnection.js

```diff
@@ -0,0 +1,760 @@
+/* global __filename, RTCIceGatherer, RTCIceTransport */
+
+import { getLogger } from 'jitsi-meet-logger';
+import yaeti from 'yaeti';
+
+import { InvalidStateError } from './errors';
+
+const logger = getLogger(__filename);
+
+const RTCSignalingState = {
+    stable: 'stable',
+    haveLocalOffer: 'have-local-offer',
+    haveRemoteOffer: 'have-remote-offer',
+    closed: 'closed'
+};
+
+const RTCIceGatheringState = {
+    new: 'new',
+    gathering: 'gathering',
+    complete: 'complete'
+};
+
+/**
+ * RTCPeerConnection shim for ORTC based endpoints (such as Edge).
+ *
+ * The interface is based on the W3C specification of 2015, which matches
+ * the implementation of Chrome nowadays:
+ *
+ *   https://www.w3.org/TR/2015/WD-webrtc-20150210/
+ */
+export default class ortcRTCPeerConnection extends yaeti.EventTarget {
+    /**
+     */
+    constructor(pcConfig) {
+        super();
+
+        logger.debug('constructor() pcConfig:', pcConfig);
+
+        // Closed flag.
+        this._closed = false;
+
+        // Create a RTCIceGatherer.
+        this._iceGatherer = this._createIceGatherer(pcConfig);
+
+        // RTCPeerConnection iceGatheringState.
+        // NOTE: This should not be needed, but Edge does not implement
+        // iceGatherer.state.
+        this._iceGatheringState = RTCIceGatheringState.new;
+
+        // Create a RTCIceTransport.
+        this._iceTransport = this._createIceTransport(this._iceGatherer);
+
+        // Local RTCSessionDescription.
+        this._localDescription = null;
+
+        // Set of local MediaStreams.
+        this._localStreams = new Set();
+
+        // Remote RTCSessionDescription.
+        this._remoteDescription = null;
+
+        // Set of remote MediaStreams.
+        this._remoteStreams = new Set();
+
+        // RTCPeerConnection signalingState.
+        this._signalingState = RTCSignalingState.stable;
+    }
+
+    /**
+     * Gets the current RTCSignalingState.
```

Comment:
missing @type documentation:
`@type {RTCSignalingState}`
please apply to all methods